internal/storage/bucket_handle.go

```diff
@@ -537,8 +537,48 @@ func (bh *bucketHandle) DeleteFolder(ctx context.Context, folderName string) (er
 }
 
 func (bh *bucketHandle) MoveObject(ctx context.Context, req *gcs.MoveObjectRequest) (*gcs.Object, error) {
```

Comment:
How are we limiting this to only be called for HNS buckets? Move object will not work for non-HNS.

I looked at the other PR #2831 but didn't see it.

Reply:
This condition will come when we do fs level integration. e.g. refer [here](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/fs.go#L1989)